source/tests/system/nirfmxtdscdma_driver_api_tests.cpp

```diff
@@ -64,6 +72,565 @@ TEST_F(NiRFmxTDSCDMADriverApiTests, Init_Close_Succeeds)
+  EXPECT_SUCCESS(session, client::cfg_iq_power_edge_trigger(stub(), session, "", "0", IQ_POWER_EDGE_TRIGGER_SLOPE_RISING, -20.00, 0.00, TRIGGER_MINIMUM_QUIET_TIME_MODE_AUTO, 16e-6, IQ_POWER_EDGE_TRIGGER_LEVEL_TYPE_RELATIVE, true));
+  EXPECT_SUCCESS(session, client::select_measurements(stub(), session, "", MEASUREMENT_TYPES_OBW, true));
+  EXPECT_SUCCESS(session, client::obw_cfg_sweep_time(stub(), session, "", OBW_SWEEP_TIME_AUTO_TRUE, 660e-6));
+  EXPECT_SUCCESS(session, client::obw_cfg_averaging(stub(), session, "", OBW_AVERAGING_ENABLED_FALSE, 10, OBW_AVERAGING_TYPE_RMS));
+  EXPECT_SUCCESS(session, client::initiate(stub(), session, "", ""));
+
+  const auto obw_fetch_measurement_response = client::obw_fetch_measurement(stub(), session, "", 10.00);
+  const auto obw_fetch_spectrum_response = client::obw_fetch_spectrum(stub(), session, "", 10.00);
+
+  EXPECT_SUCCESS(session, obw_fetch_measurement_response);
+  EXPECT_NE(0.0, obw_fetch_measurement_response.occupied_bandwidth());
+  EXPECT_NE(0.0, obw_fetch_measurement_response.absolute_power());
+  EXPECT_NE(0.0, obw_fetch_measurement_response.start_frequency());
+  EXPECT_NE(0.0, obw_fetch_measurement_response.stop_frequency());
+  EXPECT_SUCCESS(session, obw_fetch_spectrum_response);
+  EXPECT_NE(0.0, obw_fetch_spectrum_response.x0());
+  EXPECT_NE(0.0, obw_fetch_spectrum_response.dx());
+  EXPECT_NE(0, obw_fetch_spectrum_response.spectrum_size());
+  EXPECT_NE(0, obw_fetch_spectrum_response.spectrum().size());
+  EXPECT_NE(0.0, obw_fetch_spectrum_response.spectrum(0));
+}
+
+// The following test is disabled because the simulated device session always returns an error:
+// '-357830: Burst Timing Detection Failed. 0 midamble(s) found.'
+TEST_F(NiRFmxTDSCDMADriverApiTests, DISABLED_PvtFromExample_FetchData_DataLooksReasonable)
+{
+  auto session = init_session(stub(), PXI_5663E);
+  EXPECT_SUCCESS(session, client::cfg_frequency_reference(stub(), session, "", FREQUENCY_REFERENCE_SOURCE_ONBOARD_CLOCK, 10E+6));
+  EXPECT_SUCCESS(session, client::cfg_rf(stub(), session, "", 1.91E+9, 0.00, 0.00));
+  EXPECT_SUCCESS(session, client::cfg_iq_power_edge_trigger(stub(), session, "", "0", IQ_POWER_EDGE_TRIGGER_SLOPE_RISING, -20.00, 0.00, TRIGGER_MINIMUM_QUIET_TIME_MODE_AUTO, 16E-6, IQ_POWER_EDGE_TRIGGER_LEVEL_TYPE_RELATIVE, true));
+  EXPECT_SUCCESS(session, client::select_measurements(stub(), session, "", MEASUREMENT_TYPES_PVT , true));
+  EXPECT_SUCCESS(session, client::cfg_midamble_shift(stub(), session, "", MIDAMBLE_AUTO_DETECTION_MODE_MIDAMBLE_SHIFT, 16, 8));
+  EXPECT_SUCCESS(session, client::pvt_cfg_measurement_method(stub(), session, "", PVT_MEASUREMENT_METHOD_NORMAL));
+  EXPECT_SUCCESS(session, client::pvt_cfg_averaging(stub(), session, "", PVT_AVERAGING_ENABLED_FALSE, 10, PVT_AVERAGING_TYPE_RMS));
+  EXPECT_SUCCESS(session, client::initiate(stub(), session, "", ""));
+
+  const auto pvt_fetch_measurement_status_response = client::pvt_fetch_measurement_status(stub(), session, "", 10.00);
```

Comment:
@ThangamV-NI - First, some background: When adding RFmx personalities to ni/grpc-device we have been adding spot tests that are based on the C examples for the personality.  These tests use simulated devices (a PXI-5663E), and run as part of PR/CI builds of the ni/grpc-device repository.

But I can't get this test, based on the PvT example, to work.  Everything up to this point seems to work (no errors returned) but these pvt_fetch functions always return an error: "-357830: Burst Timing Detection Failed. 0 midamble(s) found.".  The goal of these tests is simply to verify that the grpc-device connection to RFmx TD-SCDMA is working, rather than verify that RFmx TD-SCDMA itself is working, so I'm considering commenting out everything from here to the end of the test and re-enabling the test, but it would be nice if we could call these functions without error.  Do you know what might be going wrong, or have advice on what to try?

Reply:
Hi, the error from Tdscdma Pvt Fetch is expected on simulated hardware. You need actual hardware with proper waveform generation to make the Pvt Fetch to work. So, you can comment out the piece of tests.

Reply:
Thanks @ThangamV-NI, that is reassuring.

I decided to leave this test as is rather than comment out the code that we expect to fail and enable the test.  This way it still compiles that code (which helps verify that the proto file is correct) but since the test is disabled the test output doesn't give the false impression that the test fully passed.